store/src/java/com/zimbra/util/ExternalVolumeReader.java

```diff
@@ -0,0 +1,202 @@
+            JSONArray currentJsonArray = currentJsonObject.getJSONArray("server/stores");
+
+            // Step 4: Append current JSON state array with new volume entry
+            currentJsonArray.put(volExtInfoObj);
+
+            // Step 5: Overwrite Appended current JSON state array with current JSON state Object
+            currentJsonObject.put("server/stores", currentJsonArray);
+
+            // Step 6: Set ldap attribute
+            provisioning.getLocalServer().setServerExternalStoreConfig(currentJsonObject.toString());
+        }
+        catch (JSONException e) {
+            // LOG.error("Error while processing ldap attribute ServerExternalStoreConfig", e);
+            throw e;
+        }
+    }
+
+    public void ModifyServerProperties(VolumeInfo volInfo) throws JSONException, ServiceException {
+        // LOG.info("addToServerProperties, volID : []");
+        // VolumeExternalInfo volExtInfo = volInfo.getVolumeExternalInfo();
+        try {
+            // Step 1: Fetch current JSON state object and current JSON state array
+            String serverExternalStoreConfigJson = provisioning.getLocalServer().getServerExternalStoreConfig();
+            JSONObject currentJsonObject = new JSONObject(serverExternalStoreConfigJson);
+            JSONArray currentJsonArray = currentJsonObject.getJSONArray("server/stores");
+
+            // Step 2: Create new/empty updated JSON state array
+            JSONArray updatedJsonArray = new JSONArray();
+
+            // Step 3: Do Modification and Copy currentJsonArray to updatedJsonArray
+            for (int i = 0; i < currentJsonArray.length(); i++) {
+                JSONObject tempJsonObj = currentJsonArray.getJSONObject(i);
+                if (volInfo.getId() != tempJsonObj.getInt("volumeId")) {
+                    updatedJsonArray.put(tempJsonObj);
+                }
+                else {
+                    tempJsonObj.put("name", volInfo.getName());
+                    updatedJsonArray.put(tempJsonObj);
+                }
+            }
+
+            // Step 4: Copy updatedJsonArray to currentJsonArray
+            currentJsonArray = updatedJsonArray;
+
+            // Step 5: Copy updatedJsonArray to currentJsonObject
+            currentJsonObject.put("server/stores", currentJsonArray);
+
+            // Step 6: Set ldap attribute
+            provisioning.getLocalServer().setServerExternalStoreConfig(currentJsonObject.toString());
+        }
+        catch (JSONException e) {
+            // LOG.error("Error while processing ldap attribute ServerExternalStoreConfig", e);
+            throw e;
+        }
+    }
+
+
+    /**
+     * Utility method for flat json to properties key-value
+     * @param properties
+     * @param flatJsonObject
+     * @throws JSONException
+     */
+    protected void convertFlatJsoObjectToProperties(Properties properties, JSONObject flatJsonObject) throws JSONException {
+        for (Iterator it = flatJsonObject.keys(); it.hasNext(); ) {
```

Comment:
add null check for flatJsonObject

Reply:
Didnt get your point ? where to add null check exactly ?

Reply:
```
if (flatJsonObject != null) {
    // your code here.
}